src/group/mls_group/create_commit.rs

```diff
@@ -46,7 +48,7 @@ impl MlsGroup {
 
         let sender_index = self.sender_index();
         // Make a copy of the current tree to apply proposals safely
-        let mut provisional_tree = self.tree().clone();
+        let mut provisional_tree = RatchetTree::new_from_public_tree_values(self.tree().borrow());
```

Comment:
This is equivalent to `&self.tree()`. Then you also don't need to `use std::borrow::Borrow;`.